daemon/main.c

```diff
@@ -512,10 +514,20 @@ post_phase_init(
         }
 
         //  Now it's time to catch signals.
+        int signals[] = { SIGTERM, SIGCHLD, SIGHUP };
+        sigset_t mask;
+        unsigned sig;
+        struct sigaction action = { 0 };
+        action.sa_handler = sigcatch;
+        sigemptyset(&mask);
+
+        for (sig = 0; sig < sizeof(signals) / sizeof(signals[0]); sig++) {
+            sigaddset(&mask, sig);
+
+            sigaction(sig, &action, NULL);
+        }
```

Comment:
Really sure about this code is what you want? Did you test it?

Reply:
```
kill -HUP $(pidof xhad)
tail /var/log/xha.log
Feb 24 15:13:46 UTC 2025 [info] Fencing is armed.
Feb 24 15:13:46 UTC 2025 [notice] The local host has transitioned to online state.
Feb 24 15:13:46 UTC 2025 [debug] HA daemon mlockall succeeded
Feb 24 15:13:47 UTC 2025 [info] SC: propose_master enter.
Feb 24 15:13:47 UTC 2025 [notice] propose_master returns TRUE.
Feb 24 15:14:36 UTC 2025 [warn] Xapimon: Xapi healthchecker (pid=466238) is hung. - []
Feb 24 15:14:36 UTC 2025 [info] Xapimon: restarting Xapi.
Feb 24 15:14:41 UTC 2025 [info] SC: propose_master enter.
Feb 24 15:14:41 UTC 2025 [notice] propose_master returns TRUE.
Feb 24 15:14:49 UTC 2025 [debug] HA daemon received SIGHUP.
```

Reply:
Ah I see what you mean, should use `signals[sig]`